src/archivist/fetcher/index.js

```diff
@@ -28,9 +29,13 @@ export default async function fetch({
     waitForElementsTimeout = config.get('fetcher.waitForElementsTimeout'),
   } = {},
 }) {
-  if (executeClientScripts) {
-    return fetchFullDom(url, cssSelectors, { navigationTimeout, language, waitForElementsTimeout });
-  }
+  try {
+    if (executeClientScripts) {
+      return await fetchFullDom(url, cssSelectors, { navigationTimeout, language, waitForElementsTimeout });
```

Comment:
With this change, the documentation is wrong. We are not returning a promise that will be fulfilled anymore, but the fulfilled promise. I am not sure of the performance impact. Is the higher-order try / catch worth that change? Shouldn't we instead chain a `.error` handler to the promise before returning it?

Reply:
I was wrong in my understanding of `return await` behaviour. This is still not blocking and won't change anything. Thanks @Ndpnt for having worked together on a test case, I learned something new today 😃